### PR TITLE
Refactor the dev-environment

### PR DESCRIPTION
**Pull request by gdelavald**
This is a proposal to remove much of the bloat of the dev environment, leaving only the very basic necessary to create apps and adds some documentation on the process.

It also adds a new gulp command to scaffold new Rocket.Chat Apps via `npm run create-app <name>` and documentation regarding it.

- [ ] Create the `deploy` task

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,15 +12,68 @@ npm start
...
-## Generating an App ID
+### Generating an App ID
 We require UUID Version 4 for IDs. To generate one for your App we recommend this site: https://www.uuidgenerator.net/version4
 
-## Logging Inside an App
+### Logging Inside an App
 Due to limitations of NodeJS's `vm` package we have had to implement a custom logger class.
 To make usage of this you can use `this.getLogger()` and then do the normal `console` style logging.
+
+## Rocket.Chat App Development
```
Comment (graywolf336) [inline README.md]: Should we add a section which explains how to use the `npm run create-app <name>`?

**Comment by graywolf336** [comment]
Also, I'm not sure if it's within the scope of this pull request or a feature we need to come back to, but I think we should have a command that is something like `npm run deploy <name>` which goes and deploys the App to the local Rocket.Chat server and enables it. :taco:

File: gulpfile.js [hunk 2]
```diff
@@ -6,32 +6,88 @@ const del = require('del');
...
+    "description": "",
+    "author": {
+        "name": "",
+        "support": ""
+    },
+    "classFile": "index.ts",
+    "iconFile": "icon.jpg"
+}
+`, 'utf8');
+        fs.writeFileSync(`./apps/${ argv.name }/icon.jpg`, '', 'binary');
```
[the remark this answers is not in the document]
Comment (gdelavald) [reply gulpfile.js]: The point is having the folder ready and error-free from the get-go. We could add a default image perhaps?

[the remark this answers is not in the document]
Comment (graywolf336) [reply gulpfile.js]: I am thinking so, almost need a default image that is acceptable as an "unicon'd" App. Like how there are default avatars for users who have yet to upload their profile image on various services..

File: README.md [hunk 3]
```diff
@@ -12,15 +12,72 @@ npm start
...
+
+The minimum necessary to create a Rocket.Chat App is having an `app.json` file and a `.ts` class which extends the `App` class and an icon.
+
+### npm run create-app
+
+The development tools provide a command to quickly scaffold a new Rocket.Chat App, simply run `npm run create-app "<App Name>"` and a new folder will be created inside the `apps` folder with an `app.json` file containing the basic information of your app (please remember to update the UUID), a dummy `icon.jpg` image and the initial implementation of the App on `index.ts`.
```
Comment (graywolf336) [inline README.md]: What do you think about generating a new uuid for them? https://www.npmjs.com/package/uuid4

File: gulpfile.js [hunk 4]
```diff
@@ -6,32 +6,100 @@ const del = require('del');
...
 
 gulp.task('clean-generated', function _cleanTypescript() {
-    return del(['./dist/**', './.server-dist/**']);
+    return del(['./dist/**']);
 });
 
-gulp.task('clean-fontello', function _cleanFontello() {
-    return del(['./.site/fontello/**']);
+gulp.task('create-app', function _createNewApp() {
+    const slugifiedName = slugify(argv.name);
```
Comment (graywolf336) [inline gulpfile.js]: Should this be moved down below the if statement?

File: gulpfile.js [hunk 5]
```diff
@@ -6,32 +6,100 @@ const del = require('del');
...
+    if(typeof argv.name != 'string' || argv.name === undefined) {
+      gutil.log(gutil.colors.red(figures.cross),  `Please use ${ gutil.colors.cyan('npm run create-app <name>') } to create a new Rocket.Chat App`);
+      throw new Error('Incorrect usage of create-app command');
+    }
+    if(fs.existsSync(`./apps/${ slugifiedName }`)) {
+      gutil.log(gutil.colors.red(figures.cross),  `${ gutil.colors.cyan(argv.name) } already exists in the apps folder`);
```
Comment (graywolf336) [inline gulpfile.js]: The folder already exists, so shouldn't we return out?

File: README.md (hunk 3 above)
Comment (gdelavald) [reply README.md]: Even better.

File: gulpfile.js (hunk 5 above)
Comment (graywolf336) [reply gulpfile.js]: Honestly, I personally would rather us have a `return` that way we exit as soon as possible and thus the flow is easier to understand at a glance.

File: gulpfile.js [hunk 6]
```diff
@@ -6,32 +6,93 @@ const del = require('del');
...
 
-gulp.task('clean-fontello', function _cleanFontello() {
-    return del(['./.site/fontello/**']);
+gulp.task('create-app', function _createNewApp() {
+    if(typeof argv.name != 'string' || argv.name === undefined) {
+        gutil.log(gutil.colors.red(figures.cross),  `Please use ${ gutil.colors.cyan('npm run create-app <name>') } to create a new Rocket.Chat App`);
+        throw new Error('Incorrect usage of create-app command');
+    }
+    if(argv.name.match(/[1-9]/) != null) {
+        gutil.log(gutil.colors.red(figures.cross),  `App name can't contain letters`);
```
Comment (graywolf336) [inline gulpfile.js]: Don't you mean numbers? Lol

Comment (gdelavald) [reply gulpfile.js]: Oops!

File: gulpfile.js [hunk 7]
```diff
@@ -6,32 +6,97 @@ const del = require('del');
...
+        return;
+    }
+    try {
+        fs.writeFileSync(`./apps/${ slugifiedName }/app.json`,
+`{
+    "id": "${ generateId() }",
+    "name": "${ argv.name }",
+    "nameSlug": "${ slugifiedName }",
+    "version": "0.0.1",
+    "requiredApiVersion": "^${ apiVersion }",
```
[the remark this answers is not in the document]
Comment (gdelavald) [reply gulpfile.js]: It really depends on our versioning management, I expect at this stage to have things breaking more easily, so a caret could be dangerous in some cases.

**Pull request merged**